### PR TITLE
feat: enhance test error UI

### PR DESCRIPTION
**Pull request by emilyrohrbough**
- Apart of #21398 - This portion updates the UI for better error handling in `cy.session()`. Next step will be to update how `cy.session()` captures and reports these errors. It was getting too big when doing the work all-in-one.

### User facing changelog
I wouldn't necessarily call these note-worthy changes for the changelog. Addressed in [#24266](https://github.com/cypress-io/cypress/pull/24266).

- Enhancements were made to improve the UI around reporter test errors in the Reporter.

### User Facing Changing
Review Percy screenshots to see the full impact of changes.

(edit: latest push tweaked these a bit, but these are the general look. See Percy!)

**Test Error:**
Screen Shot 2022-10-14 at 2 41 47 PM

**Recovered In-Test Error (currently only observed when using `cy.session()`)**
Screen Shot 2022-10-14 at 2 42 54 PM

### PR Tasks
- [x] Have tests been added/updated?
- [x] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [n/a] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: packages/reporter/src/commands/command.tsx [hunk 1]
```diff
@@ -258,6 +259,80 @@ interface Props {
...
 }
 
+const CommandDetails = observer(({ model, groupId, aliasesWithDuplicates }) => (
+  <span className={cs('command-info')}>
+    <span className='command-method'>
+      <span>
+        {model.event && model.type !== 'system' ? `(${displayName(model)})` : displayName(model)}
+      </span>
+    </span>
+    {!!groupId && model.type === 'system' && model.state === 'failed' && <StateIcon aria-hidden className="failed-indicator" state={model.state}/>}
```
Comment (emilyrohrbough) [inline packages/reporter/src/commands/command.tsx]: Note for Monday -- this is where the fail-indicator class came from -- verify you don't need to re-add `vertical-align: middle`

**Review by AtofStryker — COMMENTED** [review]
Is the goal of `isRecovered` to eventually be used outside of `cy.session`? Does that mean session validation fails but the rest of the test continues?

**Comment by emilyrohrbough** [comment]
> Is the goal of isRecovered to eventually be used outside of `cy.session`?

Not necessarily, this is just cleaning up how this is currently implemented. I could see this potentially being used in the future by other "workflow-like" commands but there are no plans ATM that I am away of to create others commands of this style. Think of this as a try-catch-finally or try-catch-try_something_else.

> Does that mean session validation fails but the rest of the test continues?

Yes, it does. In the context of `cy.session()` this only apply in this flow:
```restore session -> validate session -> FAIL validation -> recreate session -> validate session -> if pass continue to next command OR if fail, fail test```

**Review by AtofStryker — APPROVED** [review]
wonder if we want to name the `isRecovered` to something session specific until it is used more generically later? Either way changes look good on my end!

File: packages/reporter/src/commands/command-model.ts [hunk 2]
```diff
@@ -148,7 +149,14 @@ export default class Command extends Instrument {
...
 
-    this.err.update(props.err)
+    if (props.err) {
+      if (!this.err) {
+        this.err = new Err(props.err)
+      } else {
+        this.err.update(props.err)
+      }
+    }
+
```
Comment (mjhenkes) [inline packages/reporter/src/commands/command-model.ts:159]: Would it make sense to share this logic with the attempt-model?

**Review by mjhenkes — APPROVED** [review]

**Pull request merged**